platforms/software/ivy/src/main/java/org/gradle/api/publish/ivy/internal/artifact/DerivedIvyArtifact.java

```diff
@@ -29,34 +33,40 @@ public class DerivedIvyArtifact extends AbstractIvyArtifact {
     private final IvyArtifact original;
     private final PublicationInternal.DerivedArtifact derived;
 
-    public DerivedIvyArtifact(IvyArtifact original, PublicationInternal.DerivedArtifact derived, TaskDependencyFactory taskDependencyFactory) {
-        super(taskDependencyFactory);
+    public DerivedIvyArtifact(
+        IvyArtifact original,
+        PublicationInternal.DerivedArtifact derived,
+        TaskDependencyFactory taskDependencyFactory,
+        ProviderFactory providerFactory,
+        ObjectFactory objectFactory
+    ) {
+        super(taskDependencyFactory, providerFactory, objectFactory);
         this.original = original;
         this.derived = derived;
     }
 
     @Override
-    protected String getDefaultName() {
+    protected Provider<String> getDefaultName() {
         return original.getName();
     }
 
     @Override
-    protected String getDefaultType() {
-        return getFileExtension(getFile().getName());
+    protected Provider<String> getDefaultType() {
+        return Providers.of(getFileExtension(getFile().getName()));
```

Comment:
🤔 I wonder if it's ok to use these static factory methods or we should use `ProviderFactory`

Reply:
Well, I expect a sanity check to catch things like those. Or a comment on the static factory methods. Since there are neither of them, I guess it's fine